crates/re_viewport/src/context_menu/mod.rs

```diff
@@ -173,6 +174,11 @@ fn show_context_menu_for_selection(ctx: &ContextMenuContext<'_>, ui: &mut egui::
 
         should_display_separator |= any_action_displayed;
     }
+
+    // nothing was shown, make sure the context menu isn't empty
+    if !should_display_separator {
```

Comment:
[not actionable] this method is getting weirder, heh. Last time I already was staring at it trying to figure out how to express this whole thing simpler, but I'm still a bit at a loss :shrug:

Reply:
Yeah, that flag has double-duty. I started by adding a new `was_anything_added` flag, and it turned out to be implemented exactly like `should_display_separator`. Let's see if some renaming may help here.